homeassistant/components/sensor/switcher_kis.py

```diff
@@ -0,0 +1,264 @@
+"""Home Assistant Switcher Component.
+
+For controlling the Switcher Boiler Device (https://www.switcher.co.il/).
+Please follow configuring instructions.
+
+For use of type hints in comments for mypy checker:
+- Disabled pylint's 'W0611' (unused-import) warning.
+- Noqa'd flake8's 'F401' error (imported but unused).
+
+For pylint's incorrectly reports of no 'getLogger' in module 'logging':
+- Disabled pylint's 'E0611' (no-name-in-module) warning.
+
+Disabled pylint's 'C0326' (bad-whitespace).
+It contradicts flake8's 'E252'(missing whitespace around parameter equals).
+
+Author: Tomer Figenblat
+
+This cannot be configured as a sensor platform,
+Please follow the instruction of configuring the switcher_kis component.
+"""
+
+# pylint: disable=bad-whitespace,unused-import,no-name-in-module
+from asyncio import TimeoutError as AsyncioTimeoutError
+from asyncio import wait_for
+from logging import getLogger
+from typing import Any, Awaitable, Callable, Dict, List, Optional  # noqa F401
+
+from aioswitcher.consts import DISABLE_SCHEDULE, ENABLE_SCHEDULE
+from aioswitcher.devices import SwitcherV2Device  # noqa F401
+from aioswitcher.schedules import (SwitcherV2Schedule,
+                                   calc_next_run_for_schedule)
+from aioswitcher.swapi import delete_schedule, disable_enable_schedule
+
+from homeassistant.components.sensor import ENTITY_ID_FORMAT as SENSOR_FORMAT
+from homeassistant.components.switcher_kis import (DISCOVERY_CONFIG,
+                                                   DISCOVERY_DEVICE,
+                                                   DISCOVERY_SCHEDULES)
+from homeassistant.components.switcher_kis import \
+    ENTITY_ID_FORMAT as SWITCHER_KIS_FORMAT
+from homeassistant.components.switcher_kis import \
+    async_register_sensor_entities
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_ICON, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+
+DEPENDENCIES = ['switcher_kis']
+
+_LOGGER = getLogger(__name__)
+
+ENTITY_ID_FORMAT = SENSOR_FORMAT.format(SWITCHER_KIS_FORMAT)
+
+STATE_NOT_ENABLED = "Not enabled"
+STATE_NOT_CONFIGURED = "Not configured"
+STATE_NOT_RUNNING = "Not running"
+
+SCHEDULE_NAME_FORMAT = '{}_schedule{}'
+SCHEDULE_FRIENDLY_NAME_FORMAT = "{} Schedule {}"
+
+ATTR_ENABLED = 'enabled'
+ATTR_RECURRING = 'recurring'
+ATTR_START_TIME = 'start_time'
+ATTR_END_TIME = 'end_time'
+ATTR_DURATION = 'duration'
+ATTR_DAYS = 'days'
+ATTR_CONFIGURED = 'configured'
+ATTR_SCHEDULE_ID = 'schedule_id'
+
+
+async def async_setup_platform(hass: HomeAssistant, config: Dict,
+                               async_add_entities: Callable,
+                               discovery_info: Dict) -> None:
+    """Set up the switcher platform for the sensor component."""
+    if discovery_info.get(DISCOVERY_CONFIG):
```

Comment:
discovery should be in `__init__` and switch/sensor components created for each device discovered

Reply:
First things first - Thank you for your reviews, I'm going through all of them right and will push the fixed version today.

I didn't quite understand this review,
If I'm changing the file structure as suggested in the previous review,
There's not going to be a `__init__.py` file for each platform, just one for the component.
The new file structure will be:
- switcher_kis
  - __ init__.py
  - sensor.py
  - services.yaml
  - switch.py

Can you please clarify for me?

BTW
This is my first PR here, so I'm sorry if I'm a little slow at understanding.
:-)

Reply:
That's correct.